### PR TITLE
Integrate Minion into the cluster

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Integrate PinotHelixTaskResourceManager and PinotTaskManager into ControllerStarter
Let MinionClusterIntegrationTest extends HybridClusterIntegrationTest and run all query tests as well
Added tests for PinotHelixResourceManager APIs in MinionClusterIntegrationTest

### CODE REVIEW

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java [hunk 1]
```diff
@@ -108,56 +90,76 @@ public void testStopAndResumeJobQueue()
...
     Assert.assertEquals(stoppedTaskCount, 4);
 
-    // Resume the job queue
+    // Resume the task queue
     _pinotHelixTaskResourceManager.resumeTaskQueue(TestTaskGenerator.TASK_TYPE);
 
-    // Wait at most 60 seconds for all tasks completed
+    // Wait at most 60 seconds for all tasks COMPLETED
     int completedTaskCount = 0;
     while (completedTaskCount != 4) {
```
Comment (jfim) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: This loop doesn't check the end time.

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java [hunk 2]
```diff
@@ -108,56 +90,76 @@ public void testStopAndResumeJobQueue()
...
       stoppedTaskCount = 0;
       taskStates = _pinotHelixTaskResourceManager.getTaskStates(TestTaskGenerator.TASK_TYPE);
       for (TaskState taskState : taskStates.values()) {
-        if (taskState.equals(TaskState.STOPPED)) {
+        if (taskState == TaskState.STOPPED) {
           stoppedTaskCount++;
         }
       }
     }
     Assert.assertEquals(stoppedTaskCount, 4);
```
Comment (jfim) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: Add a message to this assert saying that all tasks did not stop within the expected time, to make it easier to analyze the assertion failure. Do it for all assertions that are time-based in this test.

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java [hunk 3]
```diff
@@ -18,76 +18,58 @@
...
   private PinotHelixTaskResourceManager _pinotHelixTaskResourceManager;
   private PinotTaskManager _pinotTaskManager;
-  private MinionStarter[] _minionStarters = new MinionStarter[NUM_WORKERS];
 
   @BeforeClass
   public void setUp()
       throws Exception {
-    // Start a ZooKeeper instance.
-    _zookeeperInstance = ZkStarter.startLocalZkServer(ZK_PORT);
+    super.setUp();
```
Comment (jfim) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: Add a comment that the parent `setUp` sets up ZK, kafka, controller, brokers and servers.

Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: Done.

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java (hunk 2 above)
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: Done.

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/MinionClusterIntegrationTest.java]: Oh, good catch!

**Pull request merged**